finagle-thrift/src/test/scala/com/twitter/finagle/thrift/EndToEndTest.scala

```diff
@@ -519,6 +520,32 @@ class EndToEndTest extends FunSuite with ThriftTest with BeforeAndAfter {
 
     Await.ready(ss.close())
   }
+
+  test("clientId is not sent and prep stats are not recorded when TTwitter upgrading is disabled") {
+    val pf = Protocols.binaryFactory()
+    val iface = new BServiceImpl {
+      override def someway(): Future[Void] = {
+        ClientId.current.map(_.name) match {
+          case Some(name) => Future.exception(presentClientIdEx)
+          case _ => Future.Void
+        }
+      }
+    }
+    val server = Thrift.server
+      .withProtocolFactory(pf)
+      .serveIface(new InetSocketAddress(InetAddress.getLoopbackAddress, 0), iface)
+
+    val sr = new InMemoryStatsReceiver()
+    val client = Thrift.client
+      .configured(Stats(sr))
+      .withProtocolFactory(pf)
+      .withClientId(ClientId("aClient"))
+      .withNoAttemptProtocolUpgrade
+      .newIface[B.ServiceIface](server)
+
+    assert(Await.result(client.someway()) == null)
```

Comment:
please add a short timeout to this await to pre-empt indefinitely blocking tests.